api/notd/token_metadata_processor.py

```diff
@@ -126,13 +132,17 @@ async def retrieve_token_metadata(self, registryAddress: str, tokenId: str) -> R
         try:
```

Comment:
the first thing you should do is see if the contract adheres to 721 or 1155 and then chose the function to call based on that rather than the try except structure you have below